sys/ecc/golay2412.c

```diff
@@ -187,99 +181,109 @@ static uint32_t golay2412_decode_symbol(uint32_t _sym_enc,
 
     /* compute syndrome vector, s = r*H^T = ( H*r^T )^T */
     s = golay2412_matrix_mul(_sym_enc, _B, 12);
-#if DEBUG_FEC_GOLAY2412
-    printf("s (syndrome vector): "); liquid_print_bitstring(s, 12); printf("\n");
-#endif
+    if (IS_ACTIVE(ENABLE_DEBUG)) {
+        printf("s (syndrome vector): ");
+        liquid_print_bitstring(s, 12);
+        printf("\n");
+    }
```

Comment:
This will fail to compile, as `liquid_print_bitstring()` is only defined with `ENABLE_DEBUG` set to `1`. Even though the compiler could optimize out the call on `ENABLE_DEBUG` being `0`, it would still check for the existence.

As `liquid_print_bitstring` is `static inline` anyway, you could just drop the `#if ENABLE_DEBUG` ... `#endif` guarding the definition. The compiler won't emit code for it, when it is never called.

Reply:
Solved for next PR.